Source/DafnyLanguageServer/Workspace/ProjectManager.cs

```diff
@@ -126,6 +127,10 @@ public ProjectManager(
 
   private const int MaxRememberedChanges = 100;
 
+  public TextReader ReadFile(Uri fileToRead) {
```

Comment:
This method here makes no sense to me. Why not use a filesystem directly?

Reply:
Your update for reading the last two characters forgot the test case when the position is given in line zero but the column is the number of characters. That fails the test slowlytypefile

Reply:
I added a fix for `slowlytypefile`, although I don't recognize the bug you describe.